### DISCUSSION TITLE
HTMX integration with Astro (Meta framework)

### DISCUSSION THREAD

**Discussion by FlorentDardenne** [bigskysoftware/htmx#1425] [Ideas]
Hi,

I discovered recently Astro and Htmx (yesterday). Two amazing tools for SSR websites ! 
An implementation htmx/Astro sample should be very interesting  package (and a good way to promote Htmx).

Regards,
Florent

**Comment by TymonZ** [comment]
It would be great!

I'm building a website using Astro and I really miss some htmx utilities.

**Comment by matiboy** [comment]
Hi,

It appears to be quite straightforward to integrate htmx into Astro; or are there specific features that you've found aren't available? Basically as long as your "backend" can take in requests and return HTML, optionally with specific headers, it should be good to work with htmx regardless of the language/framework

For Astro, this worked for me:

1. add `htmx.js` to public folder and add `<script src="/htmx.js" is:inline></script>` to your Layout.astro inside `<head>`
2. create a file called `say_hello.html.js` inside pages folder
3. add the regular htmx attributes to your page e.g. index.astro

```js
// say_hi.html.js file
export async function get({ params, request }) {
  return {
    body: 'Hello World',
  };
}
```

```html
<!-- content of index.astro's template -->
<main>
    <button hx-get="say_hello.html" hx-target="#outcome">Click me</button>
    <div id="outcome"></div>
</main>
```

[Sample demo](https://stackblitz.com/edit/github-qw8rxn?file=src%2Flayouts%2FLayout.astro)

Hope this helps

**Reply by TymonZ** [reply]
Thank you for your reply, I used your way of integrating them and it works :)

**Comment by fkromer** [comment]
@FlorentDardenne There is https://github.com/xstevenyung/astro-htmx which could be sufficient already.

**Reply by efosao** [reply]
I benchmarked Bun/Astro/HTMX vs Bun/Elysia/HTMX and the implementation using Astro was about 3-4x slower than just using Bun + Elysia and JSX or TSX templates. Bun + Hono and TSX templates was even **_faster_**, but it lagged behind Elysia in features, so I moved away from it for my experiments.

Also `hx-boost` and Astro _transitions_ API don't work well together.

You can use Astro if you need to, but it's a waste and code-bloat if all you're getting out of it is the templates.

## Links to Experiments
* https://bun.vauntly.com/posts
* https://astro.vauntly.com/posts

**Reply by ColeTownsend** [reply]
Is that stil the case?
